storage/src/backend/registry.rs

```diff
@@ -295,28 +298,51 @@ impl RegistryState {
         auth: &BearerAuth,
         connection: &Arc<Connection>,
     ) -> Result<Response> {
-        let mut form = HashMap::new();
-        form.insert("service".to_string(), auth.service.clone());
-        form.insert("scope".to_string(), auth.scope.clone());
-        form.insert("grant_type".to_string(), "password".to_string());
-        form.insert("username".to_string(), self.username.clone());
-        form.insert("passward".to_string(), self.password.clone());
-        form.insert("client_id".to_string(), REGISTRY_CLIENT_ID.to_string());
-
-        let mut headers = HeaderMap::new();
-
-        let token_resp = connection
-            .call::<&[u8]>(
-                Method::POST,
-                auth.realm.as_str(),
-                None,
-                Some(ReqBody::Form(form)),
-                &mut headers,
-                true,
-            )
-            .map_err(|e| einval!(format!("registry auth server request failed {:?}", e)))?;
-
-        Ok(token_resp)
+        let mut retries = 0;
+        let mut username = self.username.clone();
+        let mut password = self.password.clone();
+        loop {
+            let mut form = HashMap::new();
+            form.insert("service".to_string(), auth.service.clone());
+            form.insert("scope".to_string(), auth.scope.clone());
+            form.insert("grant_type".to_string(), "password".to_string());
+            form.insert("username".to_string(), username.clone());
+            form.insert("passward".to_string(), password.clone());
+            form.insert("client_id".to_string(), REGISTRY_CLIENT_ID.to_string());
+
+            let mut headers = HeaderMap::new();
+
+            let token_resp = connection
+                .call::<&[u8]>(
+                    Method::POST,
+                    auth.realm.as_str(),
+                    None,
+                    Some(ReqBody::Form(form)),
+                    &mut headers,
+                    false,
+                )
+                .map_err(|e| einval!(format!("registry auth server request failed {:?}", e)))?;
+
+            if token_resp.status() == StatusCode::OK {
+                return Ok(token_resp);
+            } else if token_resp.status() == StatusCode::FORBIDDEN && retries < MAX_RETRIES {
```

Comment:
Should be `vec![StatusCode::UNAUTHORIZED, StatusCode::FORBIDDEN].contains(&token_resp.status())`.